### ISSUE TITLE
optimizing for framebufferRetainCount

### ISSUE THREAD

**Issue by JamesHowlettLA** [BradLarson/GPUImage2#175]
I find the framebufferRetainCount can be removed with a wrapping class, using the reference count of ARC of the wrapping class. And the frame buffer pool automatically becomes ARC, and no need maintain the retain count.

Bad news is, each operation on frame buffer need one more access.

**Comment by BradLarson** [comment]
I don't believe this is correct, because there are certain situations where explicit locking / unlocking is necessary in order to preserve the contents of the framebuffer until just the point where they can be deallocated and allow for early reuse. 

A chain of same-sized filters used to capture a still photo, for example, may not allow for early recycling of a framebuffer if the framebuffer is attached to the enclosing class. That eliminates the reason I've added reusable cached framebuffers.

If you can demonstrate that simple ARC can allow for the limited-lifetime scope of the framebuffer by itself in all these cases, I'd be glad to listen to alternatives.
